lib/resque/worker.rb

```diff
@@ -192,51 +319,45 @@ def perform(job)
     # nil if no job can be found.
     def reserve
```

Comment:
This seems to be a major point of difference, since the check-one-at-a-time approach will give priority to the first queue in the list.